readme.md

```diff
@@ -8,23 +8,198 @@ See the "docs" directory for documentation and EIPs, and the "sharding" director
+
+    ```
+    ln -s /usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib /usr/local/lib/
+    ln -s /usr/local/opt/openssl/lib/libssl.1.0.0.dylib /usr/local/lib/
+    ```
+
+3. Latest Ethereum PyPI dependency not supported by Viper on macOS
+
+  * Problem: Error occurs when trying to install dependencies with `python setup.py install` such as `error: ethereum 2.3.0 is installed but ethereum==2.1.0 is required by {'viper'}` complaining that the version installed of [Ethereum from Python Package Index (PyPI)](https://pypi.python.org/pypi/ethereum/2.3.0) is not supported by Viper after upgrading to the latest version Python (i.e. Python 3.6 or 3.7)
+
+  * Solution: Uninstall old version and reinstall new version before running `python setup.py install` again
+
+    ```
+    python -m pip uninstall ethereum==2.3.0;
+    python -m pip install ethereum==2.1.0
+    ```
+
+4. Unit Test not passing due to Ethereum PyPI dependency failing
+
+ * Problem: Unit Test not passing due to error `E  KeyError: b'GENESIS_NUMBER'` in test file `sharding/tests/test_main_chain.py`
+   * References:
+     * https://github.com/ethereum/sharding/pull/51#issuecomment-355720781
+     * https://github.com/ethereum/sharding/pull/53
+
+* Solution: Do not install/reinstall Ethereum from Python Package Index (PyPI) or Viper manually (i.e. do not install with `python -m pip install ethereum==2.1.0`), as doing so does not install the dependency correctly. Instead install them using `USE_PYETHEREUM_DEVELOP=1 python setup.py develop`
+
+ * Example: If you incorrectly try to install Ethereum from PyPI manually with `python -m pip install ethereum==2.1.0`, then when you run `pip show ethereum` it shows the following:
+
+   ```shell
+   $ pip show ethereum
+   Name: ethereum
+   Version: 2.1.0
+   Summary: Next generation cryptocurrency network
+   Home-page: https://github.com/ethereum/pyethereum/
+   Author: UNKNOWN
+   Author-email: UNKNOWN
+   License: UNKNOWN
+   Location: /Users/Ls/.pyenv/versions/3.6.4rc1/lib/python3.6/site-packages
+   Requires: scrypt, pyethash, py-ecc, coincurve, PyYAML, rlp, pbkdf2, pycryptodome, repoze.lru, pysha3
+   ```
+
+ * Whereas when you correctly install Ethereum from PyPI using `USE_PYETHEREUM_DEVELOP=1 python setup.py develop` then when you run `pip show ethereum` you will notice that the value for its Location is provided, hence the reason why this approach works.
+
+   ```shell
+   $ pip show ethereum
+   Name: ethereum
+   Version: 2.1.0
+   Summary: Next generation cryptocurrency network
+   Home-page: https://github.com/ethereum/pyethereum/
+   Author: UNKNOWN
+   Author-email: UNKNOWN
+   License: UNKNOWN
+   Location: /Users/Ls/.pyenv/versions/3.6.4rc1/lib/python3.6/site-packages/ethereum-2.1.0-py3.6.egg
+   Requires: py-ecc, scrypt, pyethash, future, pbkdf2, repoze.lru, pysha3, coincurve, pycryptodome, PyYAML, rlp
+   ```
+
+5. Remove warning when running Unit Tests
```

Comment:
If it's fine to remove `pytest-catchlog` in `dev_requirements.txt`, this paragraph could be removed.